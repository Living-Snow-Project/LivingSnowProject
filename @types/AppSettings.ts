// TODO: explore making this a .d.ts
export type AppSettings = {
  name: string | undefined;
  organization: string | undefined;
  showFirstRun: boolean;
  showGpsWarning: boolean;
  showAtlasRecords: boolean;
  showOnlyAtlasRecords: boolean;
};
