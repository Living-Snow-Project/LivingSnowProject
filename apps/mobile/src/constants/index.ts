export * from "./Colors";
export * from "./Strings";
export * from "./TestIds";
