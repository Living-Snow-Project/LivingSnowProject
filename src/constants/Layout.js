import { Dimensions } from "react-native";

const { height, width } = Dimensions.get("window");

export default {
  window: {
    width,
    height,
  },
  isSmallDevice: width < 375,
};
