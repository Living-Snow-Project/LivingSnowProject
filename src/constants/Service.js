export const serviceEndpoint = 'https://snowalgaeproductionapp.azurewebsites.net';
