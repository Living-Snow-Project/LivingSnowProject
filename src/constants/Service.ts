const serviceEndpoint = "https://snowalgaeproductionapp.azurewebsites.net";
const photosBlobStorageEndpoint =
  "https://snowalgaestorage.blob.core.windows.net/photos";

export { serviceEndpoint, photosBlobStorageEndpoint };
