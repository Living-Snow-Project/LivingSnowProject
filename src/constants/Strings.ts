// for TextInput placeholder prop
const Placeholders = {
  GPS: {
    AcquiringLocation: "Looking for GPS signal",
    EnterCoordinates: "ie. 12.345678, -123.456789",
    NoPermissions:
      "Check location permissions in Settings. Enter coordinates manually.",
    NoLocation: "Could not determine location. Enter coordinates manually.",
  },
  // TODO: RecordScreen placeholders
};

const Labels = {
  TimelineScreen: {
    DownloadedRecords: "Downloaded",
    NoRecords: "No records to display",
    PendingRecords: "Pending",
  },
  StatusBar: {
    NoConnection: "No Internet Connection",
  },
};

// for Alert.alert calls
const Notifications = {
  uploadSuccess: {
    title: "Upload succeeded",
    message: "Thanks for your submission.",
  },
  uploadFailed: {
    title: "Record Saved",
    message: "We will upload it later.",
  },
  invalidCoordinates: {
    title: "Invalid GPS coordinates",
    message:
      'Coordinates must be in "lat, long" format. ie. 12.345678, -123.456789',
  },
};

export { Notifications, Placeholders, Labels };
