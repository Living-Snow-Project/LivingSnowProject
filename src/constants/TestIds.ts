const TestIds = {
  GPS: {
    gpsManualInputTestId: "gps-manual-input",
    gpsManualPressableTestId: "gps-manual-pressable",
  },
  Photos: {
    photoSelectorTestId: "photo-selector",
  },
  Pickers: {
    recordSelectorTestId: "record-type-picker",
    atlastSelectorTestId: "atlas-type-picker",
  },
  RecordScreen: {
    UploadButton: "record-screen-upload-button",
  },
  TimelineScreen: {
    RefreshControl: "timeline-screen-refresh-control",
    SettingsButton: "timeline-screen-settings-button",
    NewRecordButton: "timeline-screen-record-button",
  },
  SettingsScreen: {
    ShowGpsWarning: "settings-screen-show-gps-warning",
    ShowAtlasRecords: "settings-screen-show-atlas-records",
    ShowOnlyAtlasRecords: "settings-screen-show-only-atlas-records",
  },
  Icons: {
    SampleIcon: "sample-icon",
    SightingIcon: "sighting-icon",
    PictureIcon: "picture-icon",
    DeleteIcon: "delete-icon",
    EditIcon: "edit-icon",
    SnowIcon: "snow-icon",
  },
  RecordList: {
    EditRecordAction: "edit-record-action",
    DeleteRecordAction: "delete-record-action",
  },
};

export default TestIds;
