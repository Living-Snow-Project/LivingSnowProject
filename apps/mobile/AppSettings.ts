import { saveAppConfig } from "./src/lib/Storage";
import { AppSettings } from "./types/AppSettings";

const DefaultAppSettings: AppSettings = {
  name: undefined,
  organization: undefined,
  showFirstRun: true,
  showGpsWarning: true,
  colorMode: "light",
};

let appSettingsContext: AppSettings = DefaultAppSettings;

function getAppSettings(): AppSettings {
  return appSettingsContext;
}

function setAppSettings(
  param: AppSettings | ((appSettings: AppSettings) => AppSettings)
): AppSettings {
  if (typeof param === "function") {
    appSettingsContext = param(appSettingsContext);
  } else {
    appSettingsContext = { ...param };
  }

  saveAppConfig(appSettingsContext);

  return appSettingsContext;
}

export { DefaultAppSettings, getAppSettings, setAppSettings };
