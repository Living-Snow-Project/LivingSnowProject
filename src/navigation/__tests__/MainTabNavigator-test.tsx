import React from "react";
import { render, waitFor } from "@testing-library/react-native";
import Navigation from "../MainTabNavigator";
import { setAppSettings } from "../../../AppSettings";
import { Labels } from "../../constants/Strings";

// https://github.com/ptomasroos/react-native-scrollable-tab-view/issues/642
jest.mock("react-native/Libraries/Animated/NativeAnimatedHelper");

describe("Navigation test suite", () => {
  test("renders first run screen", () => {
    const { toJSON } = render(<Navigation />);

    expect(toJSON()).toMatchSnapshot();
  });

  test("renders timeline screen", async () => {
    setAppSettings((prev) => ({ ...prev, showFirstRun: false }));
    const { getByText, toJSON } = render(<Navigation />);

    await waitFor(() => getByText(Labels.TimelineScreen.NoRecords));
    expect(toJSON()).toMatchSnapshot();
  });
});
