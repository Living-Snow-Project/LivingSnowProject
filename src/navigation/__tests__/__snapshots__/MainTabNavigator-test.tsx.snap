// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation test suite renders first run screen 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <RNCSafeAreaProvider
    onInsetsChange={[Function]}
    style={
      Array [
        Object {
          "flex": 1,
        },
        undefined,
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "backgroundColor": "rgb(242, 242, 242)",
            "flex": 1,
          },
          undefined,
        ]
      }
    >
      <View
        pointerEvents="box-none"
        style={
          Object {
            "zIndex": 1,
          }
        }
      >
        <View
          accessibilityElementsHidden={false}
          importantForAccessibility="auto"
          onLayout={[Function]}
          pointerEvents="box-none"
          style={null}
        >
          <View
            pointerEvents="box-none"
            style={
              Object {
                "bottom": 0,
                "left": 0,
                "opacity": 1,
                "position": "absolute",
                "right": 0,
                "top": 0,
                "zIndex": 0,
              }
            }
          >
            <View
              style={
                Object {
                  "backgroundColor": "rgb(255, 255, 255)",
                  "borderBottomColor": "rgb(216, 216, 216)",
                  "flex": 1,
                  "shadowColor": "rgb(216, 216, 216)",
                  "shadowOffset": Object {
                    "height": 0.5,
                    "width": 0,
                  },
                  "shadowOpacity": 0.85,
                  "shadowRadius": 0,
                }
              }
            />
          </View>
          <View
            pointerEvents="box-none"
            style={
              Object {
                "height": 44,
                "maxHeight": undefined,
                "minHeight": undefined,
                "opacity": undefined,
                "transform": undefined,
              }
            }
          >
            <View
              pointerEvents="none"
              style={
                Object {
                  "height": 0,
                }
              }
            />
            <View
              pointerEvents="box-none"
              style={
                Object {
                  "alignItems": "stretch",
                  "flex": 1,
                  "flexDirection": "row",
                }
              }
            >
              <View
                pointerEvents="box-none"
                style={
                  Object {
                    "alignItems": "flex-start",
                    "flexBasis": 0,
                    "flexGrow": 1,
                    "justifyContent": "center",
                    "marginStart": 0,
                    "opacity": 1,
                  }
                }
              />
              <View
                pointerEvents="box-none"
                style={
                  Object {
                    "justifyContent": "center",
                    "marginHorizontal": 16,
                    "maxWidth": 718,
                    "opacity": 1,
                  }
                }
              >
                <Text
                  accessibilityRole="header"
                  aria-level="1"
                  numberOfLines={1}
                  onLayout={[Function]}
                  style={
                    Object {
                      "color": "rgb(28, 28, 30)",
                      "fontSize": 17,
                      "fontWeight": "600",
                    }
                  }
                >
                  Welcome
                </Text>
              </View>
              <View
                pointerEvents="box-none"
                style={
                  Object {
                    "alignItems": "flex-end",
                    "flexBasis": 0,
                    "flexGrow": 1,
                    "justifyContent": "center",
                    "marginEnd": 0,
                    "opacity": 1,
                  }
                }
              />
            </View>
          </View>
        </View>
      </View>
      <RNSScreenContainer
        onLayout={[Function]}
        style={
          Object {
            "flex": 1,
          }
        }
      >
        <RNSScreen
          activityState={2}
          forwardedRef={[Function]}
          pointerEvents="box-none"
          style={
            Object {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            }
          }
        >
          <View
            collapsable={false}
            style={
              Object {
                "opacity": 1,
              }
            }
          />
          <View
            accessibilityElementsHidden={false}
            closing={false}
            gestureVelocityImpact={0.3}
            importantForAccessibility="auto"
            onClose={[Function]}
            onGestureBegin={[Function]}
            onGestureCanceled={[Function]}
            onGestureEnd={[Function]}
            onOpen={[Function]}
            onTransition={[Function]}
            pointerEvents="box-none"
            style={
              Array [
                Object {
                  "display": "flex",
                  "overflow": undefined,
                },
                Object {
                  "bottom": 0,
                  "left": 0,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                },
              ]
            }
            transitionSpec={
              Object {
                "close": Object {
                  "animation": "spring",
                  "config": Object {
                    "damping": 500,
                    "mass": 3,
                    "overshootClamping": true,
                    "restDisplacementThreshold": 10,
                    "restSpeedThreshold": 10,
                    "stiffness": 1000,
                  },
                },
                "open": Object {
                  "animation": "spring",
                  "config": Object {
                    "damping": 500,
                    "mass": 3,
                    "overshootClamping": true,
                    "restDisplacementThreshold": 10,
                    "restSpeedThreshold": 10,
                    "stiffness": 1000,
                  },
                },
              }
            }
          >
            <View
              pointerEvents="box-none"
              style={
                Object {
                  "flex": 1,
                }
              }
            >
              <View
                collapsable={false}
                forwardedRef={[Function]}
                needsOffscreenAlphaCompositing={false}
                onGestureHandlerEvent={[Function]}
                onGestureHandlerStateChange={[Function]}
                style={
                  Object {
                    "flex": 1,
                    "transform": Array [
                      Object {
                        "translateX": 0,
                      },
                      Object {
                        "translateX": 0,
                      },
                    ],
                  }
                }
              >
                <View
                  style={
                    Array [
                      Object {
                        "flex": 1,
                        "overflow": "hidden",
                      },
                      Array [
                        Object {
                          "backgroundColor": "rgb(242, 242, 242)",
                        },
                        undefined,
                      ],
                    ]
                  }
                >
                  <View
                    style={
                      Object {
                        "flex": 1,
                        "flexDirection": "column-reverse",
                      }
                    }
                  >
                    <View
                      style={
                        Object {
                          "flex": 1,
                        }
                      }
                    >
                      <View
                        style={
                          Object {
                            "height": "100%",
                            "left": 0,
                            "position": "absolute",
                            "top": 0,
                            "transform": Array [
                              Object {
                                "translateY": 0,
                              },
                            ],
                            "width": "100%",
                          }
                        }
                      >
                        <RCTScrollView
                          style={
                            Object {
                              "flex": 1,
                              "marginTop": 30,
                              "paddingHorizontal": 10,
                            }
                          }
                        >
                          <View>
                            <View
                              style={
                                Object {
                                  "flexDirection": "row",
                                  "justifyContent": "center",
                                }
                              }
                            >
                              <Text />
                              <Text
                                style={
                                  Object {
                                    "fontSize": 25,
                                    "fontWeight": "bold",
                                    "marginTop": 1,
                                    "textAlign": "center",
                                  }
                                }
                              >
                                Living Snow Project
                              </Text>
                              <Text />
                            </View>
                            <Text
                              style={
                                Object {
                                  "color": "red",
                                  "fontSize": 12,
                                  "marginBottom": 20,
                                  "marginLeft": 20,
                                  "marginRight": 20,
                                  "marginTop": 20,
                                  "textAlign": "center",
                                }
                              }
                            >
                              Enter your name and the organization you are associated with, or skip and remain anonymous. You can change these at any time in the Settings tab.
                            </Text>
                            <Text
                              style={
                                Object {
                                  "fontSize": 15,
                                  "paddingLeft": 10,
                                  "paddingVertical": 2,
                                }
                              }
                            >
                              Name
                            </Text>
                            <TextInput
                              allowFontScaling={true}
                              maxLength={50}
                              onChangeText={[Function]}
                              onSubmitEditing={[Function]}
                              placeholder="Enter your name"
                              rejectResponderTermination={true}
                              returnKeyType="done"
                              style={
                                Object {
                                  "backgroundColor": "#efefef",
                                  "borderColor": "gray",
                                  "borderRadius": 4,
                                  "borderWidth": 1,
                                  "marginLeft": 10,
                                  "marginRight": 10,
                                  "paddingHorizontal": 15,
                                  "paddingVertical": 15,
                                }
                              }
                              underlineColorAndroid="transparent"
                            />
                            <Text
                              style={
                                Object {
                                  "fontSize": 15,
                                  "paddingLeft": 10,
                                  "paddingVertical": 2,
                                }
                              }
                            >
                              Organization
                            </Text>
                            <TextInput
                              allowFontScaling={true}
                              maxLength={50}
                              onChangeText={[Function]}
                              placeholder="Enter the organization you belong to (if any)"
                              rejectResponderTermination={true}
                              returnKeyType="done"
                              style={
                                Object {
                                  "backgroundColor": "#efefef",
                                  "borderColor": "gray",
                                  "borderRadius": 4,
                                  "borderWidth": 1,
                                  "marginLeft": 10,
                                  "marginRight": 10,
                                  "paddingHorizontal": 15,
                                  "paddingVertical": 15,
                                }
                              }
                              underlineColorAndroid="transparent"
                            />
                            <View
                              style={
                                Object {
                                  "alignItems": "center",
                                  "flex": 2,
                                  "justifyContent": "flex-start",
                                  "marginTop": 30,
                                }
                              }
                            >
                              <View
                                accessible={true}
                                collapsable={false}
                                focusable={true}
                                onBlur={[Function]}
                                onClick={[Function]}
                                onFocus={[Function]}
                                onResponderGrant={[Function]}
                                onResponderMove={[Function]}
                                onResponderRelease={[Function]}
                                onResponderTerminate={[Function]}
                                onResponderTerminationRequest={[Function]}
                                onStartShouldSetResponder={[Function]}
                                style={
                                  Array [
                                    Object {},
                                    Object {
                                      "opacity": 1,
                                    },
                                  ]
                                }
                                testID=""
                              >
                                <View
                                  style={
                                    Object {
                                      "backgroundColor": "lightpink",
                                      "borderRadius": 5,
                                      "height": 40,
                                      "justifyContent": "center",
                                      "width": 200,
                                    }
                                  }
                                >
                                  <Text
                                    style={
                                      Object {
                                        "fontSize": 20,
                                        "fontWeight": "bold",
                                        "textAlign": "center",
                                      }
                                    }
                                  >
                                    Let's get started!
                                  </Text>
                                </View>
                              </View>
                            </View>
                          </View>
                        </RCTScrollView>
                      </View>
                    </View>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </RNSScreen>
      </RNSScreenContainer>
    </View>
  </RNCSafeAreaProvider>
</View>
`;

exports[`Navigation test suite renders timeline screen 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <RNCSafeAreaProvider
    onInsetsChange={[Function]}
    style={
      Array [
        Object {
          "flex": 1,
        },
        undefined,
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "backgroundColor": "rgb(242, 242, 242)",
            "flex": 1,
          },
          undefined,
        ]
      }
    >
      <View
        pointerEvents="box-none"
        style={
          Object {
            "zIndex": 1,
          }
        }
      >
        <View
          accessibilityElementsHidden={false}
          importantForAccessibility="auto"
          onLayout={[Function]}
          pointerEvents="box-none"
          style={null}
        >
          <View
            pointerEvents="box-none"
            style={
              Object {
                "bottom": 0,
                "left": 0,
                "opacity": 1,
                "position": "absolute",
                "right": 0,
                "top": 0,
                "zIndex": 0,
              }
            }
          >
            <View
              style={
                Object {
                  "backgroundColor": "rgb(255, 255, 255)",
                  "borderBottomColor": "rgb(216, 216, 216)",
                  "flex": 1,
                  "shadowColor": "rgb(216, 216, 216)",
                  "shadowOffset": Object {
                    "height": 0.5,
                    "width": 0,
                  },
                  "shadowOpacity": 0.85,
                  "shadowRadius": 0,
                }
              }
            />
          </View>
          <View
            pointerEvents="box-none"
            style={
              Object {
                "height": 44,
                "maxHeight": undefined,
                "minHeight": undefined,
                "opacity": undefined,
                "transform": undefined,
              }
            }
          >
            <View
              pointerEvents="none"
              style={
                Object {
                  "height": 0,
                }
              }
            />
            <View
              pointerEvents="box-none"
              style={
                Object {
                  "alignItems": "stretch",
                  "flex": 1,
                  "flexDirection": "row",
                }
              }
            >
              <View
                pointerEvents="box-none"
                style={
                  Object {
                    "alignItems": "flex-start",
                    "flexBasis": 0,
                    "flexGrow": 1,
                    "justifyContent": "center",
                    "marginStart": 0,
                    "opacity": 1,
                  }
                }
              >
                <View
                  accessible={true}
                  collapsable={false}
                  focusable={true}
                  onBlur={[Function]}
                  onClick={[Function]}
                  onFocus={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    Array [
                      Object {
                        "left": 25,
                      },
                      Object {
                        "opacity": 1,
                      },
                    ]
                  }
                  testID="timeline-screen-settings-button"
                >
                  <Text
                    allowFontScaling={false}
                    style={
                      Array [
                        Object {
                          "color": "#ccc",
                          "fontSize": 32,
                        },
                        undefined,
                        Object {
                          "fontFamily": "ionicons",
                          "fontStyle": "normal",
                          "fontWeight": "normal",
                        },
                        Object {},
                      ]
                    }
                    testID="settings"
                  >
                    
                  </Text>
                </View>
              </View>
              <View
                pointerEvents="box-none"
                style={
                  Object {
                    "justifyContent": "center",
                    "marginHorizontal": 16,
                    "maxWidth": 590,
                    "opacity": 1,
                  }
                }
              >
                <Text
                  accessibilityRole="header"
                  aria-level="1"
                  numberOfLines={1}
                  onLayout={[Function]}
                  style={
                    Object {
                      "color": "rgb(28, 28, 30)",
                      "fontSize": 17,
                      "fontWeight": "600",
                    }
                  }
                >
                  Timeline
                </Text>
              </View>
              <View
                pointerEvents="box-none"
                style={
                  Object {
                    "alignItems": "flex-end",
                    "flexBasis": 0,
                    "flexGrow": 1,
                    "justifyContent": "center",
                    "marginEnd": 0,
                    "opacity": 1,
                  }
                }
              >
                <View
                  accessible={true}
                  collapsable={false}
                  focusable={true}
                  onBlur={[Function]}
                  onClick={[Function]}
                  onFocus={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    Array [
                      Object {
                        "right": 25,
                      },
                      Object {
                        "opacity": 1,
                      },
                    ]
                  }
                  testID="timeline-screen-record-button"
                >
                  <Text
                    allowFontScaling={false}
                    style={
                      Array [
                        Object {
                          "color": "#ccc",
                          "fontSize": 32,
                        },
                        undefined,
                        Object {
                          "fontFamily": "ionicons",
                          "fontStyle": "normal",
                          "fontWeight": "normal",
                        },
                        Object {},
                      ]
                    }
                    testID="add-circle-outline"
                  >
                    
                  </Text>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
      <RNSScreenContainer
        onLayout={[Function]}
        style={
          Object {
            "flex": 1,
          }
        }
      >
        <RNSScreen
          activityState={2}
          forwardedRef={[Function]}
          pointerEvents="box-none"
          style={
            Object {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            }
          }
        >
          <View
            collapsable={false}
            style={
              Object {
                "opacity": 1,
              }
            }
          />
          <View
            accessibilityElementsHidden={false}
            closing={false}
            gestureVelocityImpact={0.3}
            importantForAccessibility="auto"
            onClose={[Function]}
            onGestureBegin={[Function]}
            onGestureCanceled={[Function]}
            onGestureEnd={[Function]}
            onOpen={[Function]}
            onTransition={[Function]}
            pointerEvents="box-none"
            style={
              Array [
                Object {
                  "display": "flex",
                  "overflow": undefined,
                },
                Object {
                  "bottom": 0,
                  "left": 0,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                },
              ]
            }
            transitionSpec={
              Object {
                "close": Object {
                  "animation": "spring",
                  "config": Object {
                    "damping": 500,
                    "mass": 3,
                    "overshootClamping": true,
                    "restDisplacementThreshold": 10,
                    "restSpeedThreshold": 10,
                    "stiffness": 1000,
                  },
                },
                "open": Object {
                  "animation": "spring",
                  "config": Object {
                    "damping": 500,
                    "mass": 3,
                    "overshootClamping": true,
                    "restDisplacementThreshold": 10,
                    "restSpeedThreshold": 10,
                    "stiffness": 1000,
                  },
                },
              }
            }
          >
            <View
              pointerEvents="box-none"
              style={
                Object {
                  "flex": 1,
                }
              }
            >
              <View
                collapsable={false}
                forwardedRef={[Function]}
                needsOffscreenAlphaCompositing={false}
                onGestureHandlerEvent={[Function]}
                onGestureHandlerStateChange={[Function]}
                style={
                  Object {
                    "flex": 1,
                    "transform": Array [
                      Object {
                        "translateX": 0,
                      },
                      Object {
                        "translateX": 0,
                      },
                    ],
                  }
                }
              >
                <View
                  style={
                    Array [
                      Object {
                        "flex": 1,
                        "overflow": "hidden",
                      },
                      Array [
                        Object {
                          "backgroundColor": "rgb(242, 242, 242)",
                        },
                        undefined,
                      ],
                    ]
                  }
                >
                  <View
                    style={
                      Object {
                        "flex": 1,
                        "flexDirection": "column-reverse",
                      }
                    }
                  >
                    <View
                      style={
                        Object {
                          "flex": 1,
                        }
                      }
                    >
                      <View
                        style={
                          Object {
                            "backgroundColor": "#fff",
                            "flex": 1,
                          }
                        }
                      >
                        <RCTScrollView
                          refreshControl={
                            <RefreshControlMock
                              onRefresh={[Function]}
                              refreshing={false}
                            />
                          }
                          style={
                            Object {
                              "backgroundColor": "#fff",
                              "flex": 1,
                            }
                          }
                          testID="timeline-screen-refresh-control"
                        >
                          <RCTRefreshControl />
                          <View>
                            <Text
                              style={
                                Object {
                                  "marginTop": 20,
                                  "textAlign": "center",
                                }
                              }
                            >
                              No records to display
                            </Text>
                          </View>
                        </RCTScrollView>
                      </View>
                    </View>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </RNSScreen>
      </RNSScreenContainer>
    </View>
  </RNCSafeAreaProvider>
</View>
`;
